File: humanModel.m
function probability = humanModel(policy_index, theta, robot_state, policies, discount)
    [value, q_value, optimal_policies] = valueIteration(theta, discount);
    exp_q_vals = zeros(1, length(policies));
    for i = 1:length(policies)
        exp_q_vals(i) = exp(q_value(robot_state(1), robot_state(2), i));
    end

    sum = 0;
    for i = 1:length(exp_q_vals)
        if ~isnan(exp_q_vals(i))
            sum = sum + exp_q_vals(i);
        end
    end

    exp_q_vals = exp_q_vals/sum;
    probability = exp_q_vals(policy_index);

end

File: Unit_Tests/human_model.py
import numpy as np 
from value_iteration import ValueIteration

def human_model(policy_index, theta, robot_state, discount, grid_size):
    valiter = ValueIteration(grid_size)
    value, q_value, optimal_policies = valiter.value_iteration(theta, discount)
    exp_q_vals = np.zeros(len(valiter.policies))
    for i in range(len(valiter.policies)):
        exp_q_vals[i] = np.exp(q_value[robot_state[0], robot_state[1], i])

    sum_exp = 0
    for i in range(len(exp_q_vals)):
        if not np.isnan(exp_q_vals[i]):
            sum_exp += exp_q_vals[i]

    exp_q_vals /= sum_exp

    return exp_q_vals[policy_index]
